src/screens/Transactions/TransactionsScreen.tsx

```diff
@@ -1,61 +1,74 @@
-import React, { useState } from 'react';
-import { Button, Text, View } from 'react-native';
+import React, { useEffect, useState } from 'react';
+import { FlatList } from 'react-native';
 import { H2Heading } from '../../../assets/Fonts';
+import StandardLogo from '../../components/common/StandardLogo';
 import TransactionCard from '../../components/Transactions/TransactionCard';
 import {
-  AddManuallyContainer,
-  FilterRow,
-  HorizontalSpacingContainer,
-  Title,
+  getTransactionsByVendorUuid,
+  getVendorByEmail,
+} from '../../database/queries';
+import { TransactionStackScreenProps } from '../../navigation/types';
+import { Transaction } from '../../types/types';
+import { useAuthContext } from '../auth/AuthContext';
+import {
+  LogoContainer,
   TransactionsContainer,
+  TitleContainer,
+  CardContainer,
+  StartOfListView,
 } from './styles';
 
-export default function TransactionsScreen() {
-  const [transactions] = useState([
-    { id: '123123', date: '10/22/22 8:29AM PST', count: 4, price: 10.43 },
-    { id: '473844', date: '10/19/22 3:12PM PST', count: 10, price: 37.21 },
-    { id: '123827', date: '10/10/22 4:55AM PST', count: 2, price: 5.42 },
-    { id: '234724', date: '10/07/22 2:49AM PST', count: 1, price: 11.22 },
-    { id: '198343', date: '10/05/22 6:52PM PST', count: 12, price: 3.24 },
-    { id: '127428', date: '10/01/22 1:29PM PST', count: 3, price: 7.29 },
-  ]);
+export default function TransactionsScreen({
+  navigation,
+}: TransactionStackScreenProps<'TransactionsScreen'>) {
+  const [transactions, setTransactions] = useState<Transaction[]>([]);
+  const { user } = useAuthContext();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        // todo: retrieve vendor uuid
+        if (user) {
+          const vendor = await getVendorByEmail(user.email);
```

Comment:
The vendor UUID should probably also be stored in the auth context, so that it only needs to be fetched once.